pip/pep425tags.py

```diff
@@ -198,7 +247,8 @@ def get_supported(versions=None, noarch=False):
                 supported.append(('%s%s' % (impl, versions[0]), abi, arch))
 
         # Has binaries, does not use the Python API:
-        supported.append(('py%s' % (versions[0][0]), 'none', arch))
+        for arch in arches:
+            supported.append(('py%s' % (versions[0][0]), 'none', arch))
```

Comment:
`get_supported()` is used directly in 3 places:

- [index.py#L245](https://github.com/pypa/pip/blob/942a2908187ef8d0e0b8c6ecd69292d74f03bdfc/pip/index.py#L245): `len(tags)`: O(1)
- [wheel.py#L656](https://github.com/pypa/pip/blob/942a2908187ef8d0e0b8c6ecd69292d74f03bdfc/pip/wheel.py#L656): `[tags.index(c) for c in self.file_tags if c in tags]`: `self.file_tags` is the set of tag combinations in a wheel filename so won't be very large. `c in tags` complexity is O(len(tags)) = O(n). Not sure what the complexity of `tags.index(c)` is but I assume it's implemented the same way as `c in tags` and should then be O(n), so this whole statement is O(n).
- [wheel.py#L663](https://github.com/pypa/pip/blob/942a2908187ef8d0e0b8c6ecd69292d74f03bdfc/pip/wheel.py#L663): `bool(set(tags).intersection(self.file_tags))`: Set construction is O(len(tags)) = O(n) and set intersection is O(len(tags)) = O(n) so the whole thing is also O(n).

I don't see any loops or recursion in the callers that would cause any of these to be called len(tags) times since they're used on lists of wheels discovered e.g. from the index or in a cache directory. So I suppose the danger would be if an index had thousands of wheels.